File: source/weha/wikianalysis/diff/token/Movement.java
package mo.umac.wikianalysis.diff.token;

import mo.umac.wikianalysis.lexer.WikiToken;

public class Movement extends Match {
	
	public void setOldPos(int oldPos) {
		this.oldPos = oldPos;
	}

	public void setNewPos(int newPos) {
		this.newPos = newPos;
	}

	public Movement(int length, int oldPos, int newPos) {
		super(length, oldPos, newPos);
	}

	public Movement(Match match) {
		super(match.getLength(), match.getOldPos(), match.getNewPos());
	}

	public String getDescription() {
		return String.format("Mov(%d, %d, %d)", length, oldPos, newPos);
	}

	public String getLinkedDesc() {
		return String.format("Mov(%d, <a href='#o%d'>%d</a>, <a href='#n%d'>%d</a>)", length, oldPos, oldPos, newPos, newPos);
	}
	
}

File: weha/source/weha/diff/Movement.php
<?php
class Movement extends BasicEdit {

	private $length;
	
	public function getLength() {
		return $this->length;
	}

	public function getOldPos() {
		return $this->oldPos;
	}

	public function getNewPos() {
		return $this->newPos;
	}
	
	public function setOldPos($oldPos) {
		$this->oldPos = $oldPos;
	}

	public function setNewPos($newPos) {
		$this->newPos = $newPos;
	}

	public function __construct($length, $oldPos, $newPos) {
		$this->length = $length;
		$this->oldPos = $oldPos;
		$this->newPos = $newPos;
		$this->oldTokens = null;
		$this->newTokens = null;
	}

	public function getDescription() {
		return sprintf("Mov(%d, %d, %d)", $this->length, $this->oldPos, $this->newPos);
	}

	public function getContent() {
		return null;
	}

}
